### PR TITLE
Removed support for UTF-16 in TextEncoder

### LINKED ISSUE #13231: Remove support for UTF-16 in TextEncoder
Blocks #5601.

The specification was changed. This should allow us to remove the `encoder` field of the `TextEncoder` struct.

* Spec: <https://encoding.spec.whatwg.org/#textencoder>.
* Code: `components/script/dom/textencoder.rs`, `components/script/dom/webidls/TextEncoder.webidl`.
* Test: `/encoding/textencoder-constructor-non-utf.html`, (possibly) `/encoding/idlharness.html`.

### PR DESCRIPTION
**Pull request by ashrwin**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #13231  (github issue number if applicable).

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13405)

### CODE REVIEW

**Review by Ms2ger — COMMENTED** [review]
Lgtm , but will need test expectations to be fixed

@bors-servo try

**Comment by ashrko619** [comment]
@Ms2ger What changes have to be done in the test expectations ? 
There is already no UTF-8 there in the cases expected to fail

**Comment by jdm** [comment]
I see the issue - the [spec](https://encoding.spec.whatwg.org/#dom-textencoder) for the `TextEncoder` constructor no longer throws an error, but the current code does. This means that all the tests in `textencoder-constructor-non-utf.html` will continue to fail, which is why no behaviour change happened. You can run `./mach test-wpt tests/wpt/web-platform-tests/encoding/textencoder-constructor-non-utf.html` to test this locally after changing the constructor behaviour to match the specification.

File: tests/wpt/metadata/encoding/textencoder-constructor-non-utf.html.ini [hunk 1]
```diff
@@ -1,120 +1,3 @@
 [textencoder-constructor-non-utf.html]
```
Comment (Ms2ger) [inline tests/wpt/metadata/encoding/textencoder-constructor-non-utf.html.ini]: Please remove this file entirely, and squash the commits together. r=me with that done.

**Comment by ashrko619** [comment]
r=@Ms2ger

**Pull request merged**